modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/test/java/com/liferay/dynamic/data/mapping/form/renderer/internal/servlet/DDMFormTemplateContextProcessorTest.java

```diff
@@ -0,0 +1,392 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Carolina Barbosa
+ */
+@PrepareForTest(LanguageUtil.class)
+@RunWith(PowerMockRunner.class)
+public class DDMFormTemplateContextProcessorTest extends PowerMockito {
+
+	@Before
+	public void setUp() {
+		_setUpJSONFactoryUtil();
+		_setUpLanguageUtil();
+
+		_ddmFormTemplateContextProcessor = new DDMFormTemplateContextProcessor(
+			JSONUtil.put(
+				"pages", JSONFactoryUtil.createJSONArray()
+			).put(
+				"rules", JSONFactoryUtil.createJSONArray()
+			),
+			LocaleUtil.toLanguageId(_defaultLocale));
+	}
+
+	@Test
+	public void testGetDDMFormCustomField() {
+		DDMFormField ddmFormField =
+			_ddmFormTemplateContextProcessor.getDDMFormField(
+				JSONUtil.put(
+					"customProperty", 10.5
+				).put(
+					"fieldName", "Custom12345678"
+				).put(
+					"type", "custom_field"
+				));
+
+		Assert.assertEquals("Custom12345678", ddmFormField.getName());
+		Assert.assertEquals(10.5, ddmFormField.getProperty("customProperty"));
+		Assert.assertEquals("custom_field", ddmFormField.getType());
+	}
+
+	@Test
+	public void testGetDDMFormFieldsGroup() {
+		long ddmStructureId = RandomTestUtil.randomLong();
+		long ddmStructureLayoutId = RandomTestUtil.randomLong();
+
+		JSONArray rowsJSONArray = JSONUtil.putAll(
+			JSONUtil.put(
+				"columns",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"fields", JSONUtil.putAll("nestedField1")
+					).put(
+						"size", 12
+					))),
+			JSONUtil.put(
+				"columns",
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"fields", JSONUtil.putAll("nestedField2")
+					).put(
+						"size", 12
+					))));
+
+		DDMFormField ddmFormField =
+			_ddmFormTemplateContextProcessor.getDDMFormField(
+				JSONUtil.put(
+					"ddmStructureId", ddmStructureId
+				).put(
+					"ddmStructureLayoutId", ddmStructureLayoutId
+				).put(
+					"fieldName", "FieldsGroup12345678"
+				).put(
+					"nestedFields",
+					JSONUtil.putAll(
+						JSONUtil.put("fieldName", "nestedField1"),
+						JSONUtil.put("fieldName", "nestedField2"))
+				).put(
+					"rows", rowsJSONArray
+				).put(
+					"type", "fieldset"
+				).put(
+					"upgradedStructure", true
+				));
+
+		Assert.assertEquals("FieldsGroup12345678", ddmFormField.getName());
+
+		List<DDMFormField> nestedDDMFormFields =
+			ddmFormField.getNestedDDMFormFields();
+
+		Stream<DDMFormField> nestedDDMFormFieldStream =
+			nestedDDMFormFields.stream();
+
+		Set<String> nestedDDMFormFieldNames = nestedDDMFormFieldStream.map(
+			DDMFormField::getName
+		).collect(
+			Collectors.toSet()
+		);
+
+		Assert.assertEquals(
+			nestedDDMFormFieldNames.toString(), 2,
```

Comment:
@carolmariaabb, what is the purpose of use `nestedDDMFormFieldNames.toString()` here?

Reply:
Hi @natocesarrego, When I run `formatSource` without this, the following error occurs: `Add 'nestedDDMFormFieldNames.toString()' as extra parameter at the beginning of the 'Assert.assertEquals' call to provide more information to the test`

So, I understood that as a description for the test. More details about it [here](https://junit.org/junit4/javadoc/4.12/org/junit/Assert.html#assertEquals(java.lang.String,%20java.lang.Object,%20java.lang.Object)).